libs/seiscomp/math/filter/bpenv.cpp

```diff
@@ -0,0 +1,175 @@
+/***************************************************************************
+ * Copyright (C) gempa GmbH                                                *
+ * All rights reserved.                                                    *
+ * Contact: gempa GmbH (seiscomp-dev@gempa.de)                             *
+ *                                                                         *
+ * GNU Affero General Public License Usage                                 *
+ * This file may be used under the terms of the GNU Affero                 *
+ * Public License version 3.0 as published by the Free Software Foundation *
+ * and appearing in the file LICENSE included in the packaging of this     *
+ * file. Please review the following information to ensure the GNU Affero  *
+ * Public License version 3.0 requirements will be met:                    *
+ * https://www.gnu.org/licenses/agpl-3.0.html.                             *
+ *                                                                         *
+ * Other Usage                                                             *
+ * Alternatively, this file may be used in accordance with the terms and   *
+ * conditions contained in a signed written agreement between you and      *
+ * gempa GmbH.                                                             *
+ ***************************************************************************/
+
+
+#include <math.h>
+#include <seiscomp/math/filter/bpenv.h>
+
+namespace Seiscomp {
+namespace Math {
+namespace Filtering {
+
+
+// >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
+static void checkParameters(double centerFrequency, double bandwidthOctaves, int order) {
+	if (centerFrequency <= 0. )
+		throw Core::ValueException("Filter center frequency must be positive");
+	if (bandwidthOctaves <= 0. )
+		throw Core::ValueException("Filter band width must be positive");
+	if (order < 0 )
+		throw Core::ValueException("Filter order must be non-negative");
+	if (order > 10 )
+		throw Core::ValueException("Filter order > 10 makes no sense");
+}
+// <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
+
+
+
+
+// >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
+template<typename TYPE>
+BandPassEnvelope<TYPE>::BandPassEnvelope(double centerFrequency, double bandwidthOctaves, int order, double fsamp)
+: _centerFrequency(centerFrequency)
+, _bandwidthOctaves(bandwidthOctaves)
+, _order(order)
+, _samplingFrequency(0)
```

Comment:
You can with C++11 initialize the variables in the declaration. We think that this is much more readable. You don't need to change that, just a note.